SocialNetwork/SocialNetwork.Core/Interfaces/IUserService.cs

```diff
@@ -1,15 +1,13 @@
 ﻿using SocialNetwork.Core.Models;
-
 namespace SocialNetwork.Core.Interfaces;
-
-// Single Responsibility
-
 public interface IUserService
 {
-    // pagination
-    IEnumerable<User> GetAllUsers(int skip = 0, int take = 20);
-
-    Task<User> AddUser(User user);
-    Task<User> UpdateUser(User user);
-    Task DeleteUser(int id);
+    List<User> Users { get; set; }
+    List<User> GetUsers();
+    User? GetUserById(int id);
+    User? GetUserByName(string name);
+    void UpdateUser(User user);
+    void LogIn(User user);
```

Comment:
After successful login we might want to get data of this user. Therefore this method should return User so we don't call method GetUserByName to get this data